File: semester2/2/programming/lab1/task1.cpp
#include <cmath>
#include <iostream>

double countByFormula() {
    double x = 0.0, r = 0.0, d = 0.0;
    char variableNames[] = {'x','r','d'};
    double variables[] = {x,r,d};
    for (int i = 0;i <3;i++) {
        std::cout << variableNames[i] << " = ";
        std::cin >> variables[i];
    }
    x = variables[0];
    r = variables[1];
    d = variables[2];
    double a = (x * (asin(r / (pow(r, 2) + 1)) + 1 / pow(d, (x + 3))));
    std::cout << a;
    return a;
}

int main() {
    countByFormula();
    return 0;
}

File: semester2/2/programming/lab1/task1.pas
Program lab1_1;
Uses crt;

function arcsin(x:double):double;
begin
    if round(x*10000000)=10000000 then Arcsin := pi/2 else
    if round(x*10000000)=-10000000 THEN Arcsin := -pi/2 else
    arcsin := arctan(X/sqrt(1-sqr(x)));
end;


function countByFormula():double;
var x,r,d,a:double;
	i:integer;
	variableNames:array[1..3] of char = ('x', 'r', 'd');
	variables:array[1..3] of real;
begin
	for i := 1 to 3 do begin
		write(variableNames[i], ' = ');
		readln(variables[i]);
	end;
	x := variables[1];
	r := variables[2];
	d := variables[3];
	a := (x * (arcsin(r/(sqr(r) + 1))) + 1 / exp((x + 3) * ln(d)));
	writeln(a);
	countByFormula := a;
end;
Begin
	countByFormula();
End.
